torch/csrc/jit/pickler.h

```diff
@@ -183,9 +182,17 @@ class Pickler {
   // and only memoize those)
   uint32_t memo_id_ = 0;
 
-  // When arbitrary (maybe temporary) values are saved, keep them here so they
-  // can be memoized correctly
-  std::vector<c10::IValue> memoized_ivalues_;
+  // Memoization of IValues that have been written (index in table is used for
+  // BINPUT opcodes) to enable shared references
+  std::unordered_map<const void*, uint32_t> memoized_ivalue_map_;
+
+  // because we de-dup ivalues based on their raw pointer address in the above
+  // map we need to keep all the memoized values alive during the pickle.
+  // Otherwise, it is possible that a raw address gets reused for another
+  // object, and we will alias it to the old object at that address.
+  std::vector<IValue> memoized_ivalues_;
```

Comment:
This seems redundant with `memoized_ivalue_map_`, it could be 1 map of `std::unordered_map<IValue, uint32_t>` with a custom hash that calls `internalToPointer()`

Reply:
This is just a rename of what was there before. I feel like figuring out what the custom hash is doing is harder to understand than this.